pkgs/test_core/lib/src/runner/plugin/platform_helpers.dart

```diff
@@ -138,18 +138,28 @@ class _Deserializer {
       Map map => TestLocation.deserialize(map),
       _ => null,
     };
-    return Group(
-        group['name'] as String,
-        (group['entries'] as List).map((entry) {
-          var map = entry as Map;
-          if (map['type'] == 'group') return deserializeGroup(map);
-          return _deserializeTest(map)!;
-        }),
+    var entries = (group['entries'] as List).map((entry) {
+      var map = entry as Map;
+      if (map['type'] == 'group') return deserializeGroup(map);
+      return _deserializeTest(map)!;
+    }).toList();
+
+    var result = Group(group['name'] as String, entries,
         metadata: metadata,
         trace: trace,
         location: location,
         setUpAll: _deserializeTest(group['setUpAll'] as Map?),
         tearDownAll: _deserializeTest(group['tearDownAll'] as Map?));
+
+    // Now we have created the group, we can set it as the parent for all
```

Comment:
nit: consider extracting this into a method on `group`, called `reparentChildren` or similar.

Reply:
Also consider asserting the parent is null

Reply:
Moving to the constructor is perfect - and avoids doing it in two places - thanks!